internal/handlers/alumni_handler.go

```diff
@@ -0,0 +1,96 @@
+package handlers
+
+import (
+	"backend/internal/middleware"
+	"backend/internal/models"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type AlumniHandler struct {
+	db *gorm.DB
+}
+
+func NewAlumniHandler(db *gorm.DB) *AlumniHandler {
+	return &AlumniHandler{db: db}
+}
+
+func (h *AlumniHandler) Register(r *gin.RouterGroup) {
+	alumni := r.Group("/alumni")
+	{
+		// Public endpoints (require auth)
+		alumni.GET("", h.List)
+		alumni.GET("/:id", h.Get)
+
+		// Admin-only endpoints
+		admin := alumni.Group("/admin")
```

Comment:
no need to group with sepearate base URL for admin routes, just make sure the middleware `AdminRequired()` is there for the POST, PUT and DELETE endpoints